apps/system/js/app_window.js

```diff
@@ -227,10 +227,17 @@
     if (!this.isBrowser()) {
       return false;
     }
-
-    var url = new URL(this.config.url);
-    var path = url.hostname + url.pathname;
-    return path.indexOf(scope) === 0;
+    // within-scope per http://www.w3.org/TR/appmanifest/#dfn-within-scope
+    // except we also support paths
+    var withinScope;
+    var target = this.config.url;
+    if (scope) {
+      var scopeURL = new URL(scope);
+      withinScope = target.indexOf(scopeURL.origin + scopeURL.pathname) === 0;
```

Comment:
You may be right. I put this in because it was failing when scope was literally the webManifest.start_url, which cant be guaranteed to be a substring (e.g querystring params, /start.html, or whatever). But with that fixed, the indexOf should be accurate.